Add support for parsing HTML numeric entities

DerZade opened the pull request:
# Purpose / Goal
<!-- Please specify here what you're planning to achieve by this pull request and how it can help in regard of this work-space. -->
This PR adds a `htmlNumericEntities` option which adds support for parsing HTML numeric entities.


<!-- If it is a bug fix, please mention the issue number. If there is no issue has been raised but the PR directly then it should follow the template given to raise the issue. Like input, expected output, actual output etc. -->


# Type
Please mention the type of PR
<!-- choose one by changing [ ] to [x] -->
* [ ] Bug Fix
* [ ] Refactoring / Technology upgrade
* [x] New Feature

# Benchmarks

## Before
```
Running Suite: XML Parser benchmark
fxp v3 : 43885.453114704054 requests/second
fxp : 28444.884306095257 requests/second
fxp - preserve order : 35582.44370634719 requests/second
xmlbuilder2 : 12279.374624117605 requests/second
xml2js  : 17997.982194339696 requests/second
```

## After
```
Running Suite: XML Parser benchmark
fxp v3 : 56829.18072321323 requests/second
fxp : 35711.03397186936 requests/second
fxp - preserve order : 35048.53534051172 requests/second
xmlbuilder2 : 12565.714384084644 requests/second
xml2js  : 17898.357168570532 requests/second
```

amitguptagwl commented:
Thanks for your PR

I've a few suggestions
1. I believe we don't need another option. It can be done with `htmlEntities` flag only.
2. we can add an extra  property to htmlEntities object, say `num` or `hex`. And the value can be set to a function. In the loop, we can check the type if it is string replace it as today. Otherwise, run the function as per setting in the object. It'll give the chance to disable it easily.
3. both expressions (Eg `/&#([0-9]+);/`) has `+` that make it unsafe to match very long string and will impact performance too.

Let me know your thoughts

DerZade commented:
@amitguptagwl I just force pushed and updated my code to reflect your suggestions:

- Removed `+` from both expressions. Decimal is now limited to 7 chars and hexadecimal to 6 chars. (The max code point is `U+10FFFF` or `1,114,112` in decimal)
- Removed the extra `htmlNumericEntities` option 
- Added both regex and replacers to `htmlEntities`-object. I choose `num_dec` and `num_hex` as keys, since there is a named char reference `num;`
- I didn't even have to check the type for string / funcion as you suggested, since [`String.prototype.replace` already supports functions as the replacer value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#replacement).

amitguptagwl commented:
It's live now. You can see the entry in change logs

DerZade commented:
Thank you so much!
